[virt_autotest] Add extra required logs for SLE-15 host

guoxuguang opened the pull request:
refer to the current existed script data/virt_autotest/virt_logs_collector.sh, just only collect extra required logs both sles11sp4 and sles12+(i.e included sles12).

So, add extra required logs for SLE-15 VM host

- Verification run:
[gi-guest_developing-on-host_sles15sp2-xen](http://openqa.suse.de/tests/4930686)

waynechen55 commented:
Please make sure this is what you want.

For >=12,
/var/lib/libvirt/boot
/var/lib/libvirt/qemu
/var/log/libvirt/libvirtd.log
/var/log/xen/xen-boot.log (xen only)

guoxuguang commented:
> Please make sure this is what you want.
> 
> For >=12,
> /var/lib/libvirt/boot
> /var/lib/libvirt/qemu
> /var/log/libvirt/libvirtd.log
> /var/log/xen/xen-boot.log (xen only)

Great! 
Also, refer to bsc#1178048 mentioned, 
https://bugzilla.suse.com/show_bug.cgi?id=1178048#c1
There should be added /var/log/libvirt/libxl/libxl-driver.log from SLE12+(xen)

So, collect the extra required log from SLE-12+ host as below:
For >=12,
/var/lib/libvirt/boot
/var/lib/libvirt/qemu
/var/log/libvirt/libvirtd.log
/var/log/xen/xen-boot.log (xen only)
/var/log/libvirt/libxl/libxl-driver.log  (xen only)

waynechen55 commented:
> > Please make sure this is what you want.
> > For >=12,
> > /var/lib/libvirt/boot
> > /var/lib/libvirt/qemu
> > /var/log/libvirt/libvirtd.log
> > /var/log/xen/xen-boot.log (xen only)
> 
> Great!
> Also, refer to bsc#1178048 mentioned,
> https://bugzilla.suse.com/show_bug.cgi?id=1178048#c1
> There should be added /var/log/libvirt/libxl/libxl-driver.log from SLE12+(xen)
> 
> So, collect the extra required log from SLE-12+ host as below:
> For >=12,
> /var/lib/libvirt/boot
> /var/lib/libvirt/qemu
> /var/log/libvirt/libvirtd.log
> /var/log/xen/xen-boot.log (xen only)
> /var/log/libvirt/libxl/libxl-driver.log (xen only)

Suggest you collect the /var/log/libvirt/libxl folder as a whole

guoxuguang commented:
> > > Please make sure this is what you want.
> > > For >=12,
> > > /var/lib/libvirt/boot
> > > /var/lib/libvirt/qemu
> > > /var/log/libvirt/libvirtd.log
> > > /var/log/xen/xen-boot.log (xen only)
> > 
> > 
> > Great!
> > Also, refer to bsc#1178048 mentioned,
> > https://bugzilla.suse.com/show_bug.cgi?id=1178048#c1
> > There should be added /var/log/libvirt/libxl/libxl-driver.log from SLE12+(xen)
> > So, collect the extra required log from SLE-12+ host as below:
> > For >=12,
> > /var/lib/libvirt/boot
> > /var/lib/libvirt/qemu
> > /var/log/libvirt/libvirtd.log
> > /var/log/xen/xen-boot.log (xen only)
> > /var/log/libvirt/libxl/libxl-driver.log (xen only)
> 
> Suggest you collect the /var/log/libvirt/libxl folder as a whole

sure, agree on this as your mentioned.

Confirm /var/log/libvirt/libxl dir both on 12sp5 and 15sp3 xen hosts
there were still existed guest system log too 
For example:
# ls -al /var/log/libvirt/libxl
total 128
drwx------ 1 root root   172 Oct 26 23:38 .
drwx------ 1 root root    48 Oct 26 23:13 ..
-rw-r--r-- 1 root root     0 Oct 26 22:31 Domain-0.log
-rw-r--r-- 1 root root 92552 Oct 26 23:57 libxl-driver.log
-rw-r--r-- 1 root root 18195 Oct 26 23:57 sles-15-sp3-64-fv-def-net.log
-rw-r--r-- 1 root root 15398 Oct 26 23:38 sles-15-sp3-64-pv-def-net.log

alice-suse commented:
@guoxuguang @waynechen55 Please refer to http://openqa.suse.de/tests/4856934/file/guest_installation_run-virt_logs_all.tar.gz, developing xen host, the /var/log/libvirt/libxl/libxl-driver.log is in host supportconfig directory plugin-libvirt.txt. So additional log for this is NOT needed.
But why the log is not in that specific job link of the bug needs to be find out and solve. My suspect is the previous call of virt_autotest::virtual_network_utils::upload_debug_log() .

alice-suse commented:
FYI. With the just-now merged PR https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11259#issuecomment-718559792, from xen validation run, the libxl-driver.log can be uploaded in virt_logs_all.tar.gz already.

guoxuguang commented:
> FYI. With the just-now merged PR [#11259 (comment)](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11259#issuecomment-718559792), from xen validation run, the libxl-driver.log can be uploaded in virt_logs_all.tar.gz already.

sure, had removed `$libvirt_libxl_log` from `collect_extra_logs_from_host()` now

alice-suse commented:
@guoxuguang Do you think this PR can be closed now? BTW, please help explain in that bug to the dev  James Fehlig with where he can find this log in our uploaded logs in future . Thanks!

guoxuguang commented:
> @guoxuguang Do you think this PR can be closed now? BTW, please help explain in that bug to the dev James Fehlig with where he can find this log in our uploaded logs in future . Thanks!

@alice-suse ,  I do not agree to close this PR.

Because of, as my mentioned, refer to the current `collect_extra_logs_from_host()`, just only collect extra log for sles11(e.g s11sp4) and sles12 hosts without sles15 now.

So, add collect extra log for sles15 vm host as created this PR purpose - or new requirement for `collect_extra_logs_from_host()`

alice-suse commented:
> > @guoxuguang Do you think this PR can be closed now? BTW, please help explain in that bug to the dev James Fehlig with where he can find this log in our uploaded logs in future . Thanks!
> 
> @alice-suse , I do not agree to close this PR, imho.
> 
> Because of, as my mentioned, refer to the current `collect_extra_logs_from_host()`, just only collect extra log for sles11(e.g s11sp4) and sles12 hosts without sles15 now.
> 
> So, add collect extra log for sles15 vm host as created this PR purpose - or new requirement for `collect_extra_logs_from_host()`as Wayne's mentioned.



> > @guoxuguang Do you think this PR can be closed now? BTW, please help explain in that bug to the dev James Fehlig with where he can find this log in our uploaded logs in future . Thanks!
> 
> @alice-suse , I do not agree to close this PR, imho.
> 
> Because of, as my mentioned, refer to the current `collect_extra_logs_from_host()`, just only collect extra log for sles11(e.g s11sp4) and sles12 hosts without sles15 now.
> 
> So, add collect extra log for sles15 vm host as created this PR purpose - or new requirement for `collect_extra_logs_from_host()`as Wayne's mentioned.

I got your point now. I thought sle15+ host needed extra logs were already investigated/defined/added by wayne  during log enhancement design and this PR was raised only needed for libxl-driver.log. So you still want following extra logs:

For >=12,
/var/lib/libvirt/boot
/var/lib/libvirt/qemu
/var/log/libvirt/libvirtd.log
/var/log/xen/xen-boot.log (xen only)

I agree to add if they are MISSING, however please give a more careful checking before adding by EXTRA -- only missing ones from supportconfig need to be added by extra. In the same job http://149.44.176.58/tests/4913844#downloads(15sp2 xen), /var/log/libvirt/libvirtd.log is also in plugin-libvirt.txt. I do not find the following 3, so better add :
/var/lib/libvirt/boot
/var/lib/libvirt/qemu
/var/log/xen/xen-boot.log (xen only) //this is important, should be uploaded.

BTW, in 12sp5 xen host supportconfig, the libvirtd.log also uploaded in http://149.44.176.58/tests/4806030#downloads in plugin-libvirt.txt. So IMO, this one can be dropped from >SLE12  host extra logs. @waynechen55 Please double confirm. If yes, Leon, can you help make this fix together?

@waynechen55 Hope next time you can help double check log enhancement PRs during review whether the changes are really needed. Leon sometimes need your help.  I really do not have enough time to make so detailed checking for every PR. I think you task owners are able to handle perfectly. Thanks for understanding!

guoxuguang commented:
Why we had missed some extra required log on osd vm host after call `post-fail-hook()`?
we had call clean_up script during our osd test run, which had cleaned up text log file  as below:
```
The extra logs to be cleaned up are 
Going to clean up all empty and text log files in /var/log/libvirt.
Cleaned up text log file /var/log/libvirt/libvirtd.log.
Cleaned up text log file /var/log/libvirt/libxl/Domain-0.log.
Cleaned up text log file /var/log/libvirt/libxl/libxl-driver.log.
Cleaned up text log file /var/log/libvirt/libxl/sles-15-sp3-64-fv-def-net.log.
Cleaned up text log file /var/log/libvirt/libxl/sles-15-sp3-64-pv-def-net.log.
Cleaned up 5 empty and text log files in /var/log/libvirt.
Going to clean up all empty and text log files in /var/log/xen.
Cleaned up text log file /var/log/xen/qemu-dm-sles-15-sp3-64-fv-def-net.log.
Cleaned up text log file /var/log/xen/qemu-dm-sles-15-sp3-64-fv-def-net.log.1.
Cleaned up text log file /var/log/xen/qemu-dm-sles-15-sp3-64-fv-def-net.log.2.
Removed empty log file /var/log/xen/qemu-dm-sles-15-sp3-64-pv-def-net.log.
Cleaned up text log file /var/log/xen/qemu-dm-sles-15-sp3-64-pv-def-net.log.1.
Cleaned up text log file /var/log/xen/qemu-dm-sles-15-sp3-64-pv-def-net.log.2.
Cleaned up text log file /var/log/xen/xen-boot.log.
Cleaned up text log file /var/log/xen/xen-boot.prev.log.
Removed empty log file /var/log/xen/xen-hotplug.log.
Cleaned up 9 empty and text log files in /var/log/xen.
Cleaned up 14 empty and text log files in total.
Virtual machine sles-15-sp3-64-fv-def-net was running. Going to reboot it.
virsh reboot sles-15-sp3-64-fv-def-net
Domain sles-15-sp3-64-fv-def-net is being rebooted

Virtual machine sles-15-sp3-64-pv-def-net was running. Going to reboot it.
virsh reboot sles-15-sp3-64-pv-def-net
Domain sles-15-sp3-64-pv-def-net is being rebooted
```
osd test url:
http://10.160.0.207/tests/4991817#step/libvirt_virtual_network_init/19
